domains/certificates/Query.API/API/DataSyncSyncer/DataSyncSyncerWorker.cs

```diff
@@ -13,41 +17,94 @@ namespace API.DataSyncSyncer;
 internal class DataSyncSyncerWorker : BackgroundService
 {
     private readonly ILogger<DataSyncSyncerWorker> logger;
-    private readonly IEventStore eventStore;
-    private readonly string? gsrn;
+    private readonly IIntegrationEventBus integrationEventBus;
+    private readonly IDataSync dataSync;
+    private readonly List<MasterData> masterData;
+    private readonly Dictionary<string, DateTimeOffset> periodStartTimeDictionary;
 
-    public DataSyncSyncerWorker(ILogger<DataSyncSyncerWorker> logger, IEventStore eventStore,
-        MockMasterDataCollection collection)
+    public DataSyncSyncerWorker(
+        ILogger<DataSyncSyncerWorker> logger,
+        MockMasterDataCollection collection,
+        IIntegrationEventBus queue,
+        IDataSync dataSync
+    )
     {
         this.logger = logger;
-        this.eventStore = eventStore;
-        var masterData = collection.Data.FirstOrDefault();
-        gsrn = masterData?.GSRN ?? null;
+        masterData = collection.Data.ToList();
+        integrationEventBus = queue;
+        this.dataSync = dataSync;
+        periodStartTimeDictionary = masterData
+            .Where(it => !string.IsNullOrWhiteSpace(it.GSRN))
+            .ToDictionary(m => m.GSRN, m => m.MeteringPointOnboardedStartDate);
     }
 
     protected override async Task ExecuteAsync(CancellationToken stoppingToken)
     {
-        if (string.IsNullOrWhiteSpace(gsrn))
+        while (!stoppingToken.IsCancellationRequested)
         {
-            logger.LogWarning("No master data loaded. Will not produce any events");
+            await SleepToNearestHour(stoppingToken);
+
+            foreach (var data in masterData)
+            {
+                var measurements = await FetchMeasurements(data.GSRN, data.MeteringPointOwner);
+                SetNextPeriodStartTime(measurements, data.GSRN);
+
+                var integrationsEvents = MapToIntegrationEvents(measurements);
+                await integrationEventBus.Produce(stoppingToken, integrationsEvents);
+                logger.LogInformation("Produce energy measured event");
```

Comment:
This logging statement is not helpful for us in a debugging scenario. It could be either be deleted or changed to 

```suggestion
                logger.LogInformation("Produced {numEvents} energy measured events", integrationEvents.Count);
```

so we from the logs know if it is actually producing/publishing events to the bus.